docs/dev_guide/diagnostic_ref/dpct1081.rst

```diff
@@ -27,4 +27,68 @@ Suggestions to Fix
 ------------------
 
 Check the address space of the memory that <pointer variable> points to
-and replace ``address_space::global`` with the correct address space.
\ No newline at end of file
+and replace ``address_space::global`` with the correct address space.
+
+For example, this original CUDA\* code:
+
+.. code-block:: cpp
+   :linenos:
+
+   __device__ void bar1(double *d) {
+     double i = 30;
+     double d1;
+     double &d2 = *d;
+   
+     sincos(i, &d1, &d2);
+   }
+   
+   __global__ void kernel(int *p) {
+     double d;
+     bar1(&d);
+   }
+
+results in the following migrated SYCL code:
+
+.. code-block:: cpp
+   :linenos:
+
+   void bar1(double *d) {
+     double i = 30;
+     double d1;
+     double &d2 = *d;
+   
+     /*
+     DPCT1081:1: The generated code assumes that "&d2" points to the global memory
+     address space. If it points to a local or private memory address space,
+     replace "address_space::global" with "address_space::local" or
+     "address_space::private".
+     */
+     d1 = sycl::sincos(
+         i, sycl::address_space_cast<sycl::access::address_space::global_space,
+                                     sycl::access::decorated::yes, double>(&d2));
+   }
+   
+   void kernel(int *p) {
+     double d;
+     bar1(&d);
+   }
+
+which is rewritten to:
+
+.. code-block:: cpp
+   :linenos:
+
+   void bar1(double *d) {

```

Comment:
1. does sycl provide math to reduce/provide the address space?
2.  If L86 need to be more common, in case bar1() is called with ptr private_space/global_space.

Reply:
It seems spec allows generic_space here while compiler is not supported. I have opened an issue against compiler team.